Authentication/AuthenticationComponentsFactory.swift

```diff
@@ -0,0 +1,353 @@
+//
+//  AuthenticationComponentsFactory.swift
+//  Authentication
+//
+//  Created by Daniela Riesgo on 8/1/16.
+//  Copyright © 2016 Wolox. All rights reserved.
+//
+
+public protocol AuthenticationComponentsFactoryType: LoginComponentsFactory, SignupComponentsFactory {
+    
+    //var sessionService: SessionServiceType { get }
+    var initialScreen: AuthenticationInitialScreen { get }
+    
+    func createMainViewController() -> UIViewController
+    
+}
+
+/**
+    Factory that uses default implementations of the protocol
+    and stores all information needed for methods without default
+    implementation.
+    It stores the view configurations and the main view controller
+    factory method.
+ */
+public struct AuthenticationComponentsFactory: AuthenticationComponentsFactoryType {
+    
+    /// Property indicating the authentication screen to be shown the first time.
+    public let initialScreen: AuthenticationInitialScreen
+    
+    private let _mainControllerFactory: () -> UIViewController
+    private let _loginConfiguration: LoginViewConfigurationType
+    private let _signupConfiguration: SignupViewConfigurationType
+    
+    /**
+        Initializes the AuthenticationComponentsFactory
+        with the components to use.
+        
+        - Parameters:
+            - initialScreen: authentication screen to be shown
+                the first time. By default, Login.
+            - loginConfiguration: login view configuration to
+                return when asked for it.
+            - signupConfiguration: signup view configuration to
+                return when asked for it.
+            - mainControllerFactory: main view controller factory
+                method to trigger when asked for the main view
+                controller.
+    */
+    public init(initialScreen: AuthenticationInitialScreen = .Login,
+                loginConfiguration: LoginViewConfigurationType,
+                signupConfiguration: SignupViewConfigurationType,
+                mainControllerFactory: () -> UIViewController) {
+        _mainControllerFactory = mainControllerFactory
+        _loginConfiguration = loginConfiguration
+        _signupConfiguration = signupConfiguration
+        self.initialScreen = initialScreen
+    }
+    
+    /**
+         Initializes the AuthenticationComponentsFactory
+         with the components to use.
+         
+         - Parameters:
+             - initialScreen: authentication screen to be shown
+                the first time. By default, Login.
+             - logo: logo to add to the default LoginViewConfiguration.
+             - termsAndServicesURL: terms and services URL used
+                to create the default SignupViewConfiguration.
+             - mainControllerFactory: main view controller factory
+                method to trigger when asked for the main view
+                controller.
+     */
+    public init(initialScreen: AuthenticationInitialScreen = .Login,
```

Comment:
I wanted it to be a convenience init but xcode told me to erase "convenience".
Nevertheless, do you think this init is useful? Because the user will most probably (not to say always) change the color and font palettes, and so create their own view configurations.